tools/nimgrep.nim

```diff
@@ -129,9 +129,11 @@ proc processFile(pattern; filename: string) =
       stdout.flushFile()
       filenameShown = true
 
-  var buffer: string
+  # Reuse the same buffer for all files.
+  # Change to threadvar if this ever becomes multi-threaded.
+  var buffer {.global.}: string
   try:
-    buffer = system.readFile(filename)
+    system.readFile(filename, result=buffer)
```

Comment:
On my machine this commit makes `nimgrep --recursive  --ext:'nim|c' 'hello world'` about 10% faster.